### PR TITLE
feat: dashboard page

### LINKED ISSUE #133: [FEATURE]: Dashboard page
## Summary

- Users statistics 
- My recent retros (last 3months)
- Boards and sub boards with board name, nr of cards and columns, team or personal tag, avatar of users
- Calendar

## User story it relates to

#118 

### Basic example

[Figma page](https://www.figma.com/file/McDBByDqFzjYuYZ4CwVpax/UI-v2.0?node-id=106%3A2886)

### PR DESCRIPTION
**Pull request by nunocaseiro**
Relates to #133 

## Screenshots

Screenshot 2022-04-13 at 16 47 23

## Proposed Changes

  - Dashboard tiles - active members, my boards and teams
  - My recent boards - last 3 months
  - Card body with tooltips and info - board name, sub boards, responsible, team name, users
  - Calendar

This pull request is intended to close #133

### CODE REVIEW

File: frontend/components/Errors/QueryError.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+const QueryError: React.FC = ({ children }) => {
+  return (
+    <QueryErrorResetBoundary>
+      {({ reset }) => (
+        <ErrorBoundary
+          onReset={reset}
+          fallbackRender={({ resetErrorBoundary }) => (
+            <Flex css={{ my: "$24" }}>
+              There was an error!
+              <Button onClick={() => resetErrorBoundary()}>Try again</Button>
```
Comment (RubenMCCarreira) [inline frontend/components/Errors/QueryError.tsx:16]:
```suggestion
              <Button onClick={resetErrorBoundary}>Try again</Button>
```

File: package.json [hunk 2]
```diff
@@ -8,9 +8,5 @@
     "@commitlint/cli": "^16.2.3",
     "@commitlint/config-conventional": "^16.2.1",
     "husky": "^7.0.4"
-  },
-  "dependencies": {
-    "react": "^18.0.0",
-    "react-dom": "^18.0.0"
```
Comment (RubenMCCarreira) [inline package.json:14]: why do you remove this?

File: frontend/components/CardBoard/CardBody/CountCards.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  const countCards = useMemo(() => {
+    return columns.reduce((acc, column) => {
+      column.cards.forEach((card) => {
+        card.items.forEach(() => {
+          acc++;
+        });
+      });
+      return acc;
+    }, 0);
+  }, [columns]);
```
Comment (RubenMCCarreira) [inline frontend/components/CardBoard/CardBody/CountCards.tsx:19]:
```suggestion
  const countCards =  columns.reduce((acc, column) => {
  column.cards.forEach((card) => {
    acc = acc + card.items.length
  });
  return acc;
}, 0);
```

File: package.json (hunk 2 above)
Comment (nunocaseiro) [reply package.json:14]: yes ofc, this is not supposed to be there

File: frontend/components/CardBoard/CardBody/LeftArrow.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        </Flex>
+      )}
+      {!isDashboard && index !== 0 && (
+        <Flex css={{ position: "relative", size: 0 }}>
+          <Flex css={{ mt: "$2", ml: "13px", mr: "$4", position: "relative", bottom: "42px" }}>
+            <StartArrow />
+          </Flex>
+        </Flex>
+      )}
+    </Flex>
```
Comment (RubenMCCarreira) [inline frontend/components/CardBoard/CardBody/LeftArrow.tsx:28]:
```suggestion
<Flex>
      <Flex css={{ position: "relative", size: 0 }}>
        {(isDashboard || index === 0) && (
          <Flex css={{ mt: "$11", ml: "$8", mr: "$8" }}>
            <LateralUpArrow />
          </Flex>
        )}
        {!isDashboard && index !== 0 && (
          <Flex css={{ mt: "$2", ml: "13px", mr: "$4", position: "relative", bottom: "42px" }}>
            <StartArrow />
          </Flex>
        )}
      </Flex>
    </Flex>
```

File: frontend/components/Dashboard/RecentRetros/ListOfCards.tsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    return boardsByDay;
+  }, [data.pages]);
+
+  const onScroll = () => {
+    if (scrollRef.current) {
+      const { scrollTop, scrollHeight, clientHeight } = scrollRef.current;
+      if (scrollTop + clientHeight + 2 >= scrollHeight) {
+        if (fetchBoards.hasNextPage) fetchBoards.fetchNextPage();
+      }
+    }
```
Comment (RubenMCCarreira) [inline frontend/components/Dashboard/RecentRetros/ListOfCards.tsx:58]:
```suggestion
if (!scrollRef.current) return;
const { scrollTop, scrollHeight, clientHeight } = scrollRef.current;
if (scrollTop + clientHeight + 2 >= scrollHeight && fetchBoards.hasNextPage) {
  fetchBoards.fetchNextPage();
}
```

File: frontend/components/CardBoard/CardBody/CardTitle.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+          <StyledBoardTitle data-disabled={!userIsParticipating}>{title}</StyledBoardTitle>
+        </Link>
+      </Flex>
+    </Tooltip>;
+  }
+  return (
+    <Tooltip content="It’s a sub-team board. A huge team got splitted into sub teams.">
+      <StyledBoardTitle data-disabled={!userIsParticipating}>{title}</StyledBoardTitle>
+    </Tooltip>
+  );
```
Comment (RubenMCCarreira) [inline frontend/components/CardBoard/CardBody/CardTitle.tsx:41]:
```suggestion
<Tooltip content="It’s a sub-team board. A huge team got splitted into sub teams.">
  {userIsParticipating ? (
    <Flex>
      <Link key={boardId} href={{ pathname: `boards/[boardId]`, query: { boardId } }}>
        <StyledBoardTitle data-disabled={!userIsParticipating}>{title}</StyledBoardTitle>
      </Link>
    </Flex>
  ):(
    <StyledBoardTitle data-disabled={!userIsParticipating}>{title}</StyledBoardTitle>
  )}
</Tooltip>
```

File: frontend/components/CardBoard/CardBody/CardBody.tsx [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+              board={board}
+              isDashboard={isDashboard}
+              isSubBoard={isSubBoard}
+              index={index}
+              renderCardAvatars={renderCardAvatars}
+              userIsAdmin={userIsAdmin}
+            />
+          </InnerContainer>
+        </Flex>
+        {(openSubBoards === true || isDashboard) && (
```
Comment (RubenMCCarreira) [inline frontend/components/CardBoard/CardBody/CardBody.tsx:150]:
```suggestion
        {(openSubBoards || isDashboard) && (
```

File: frontend/components/CardBoard/CardBody/CardBody.tsx [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+          <CardAvatars
+            listUsers={listUsers}
+            responsible={onlyResponsible}
+            teamAdmins={teamAdmins}
+            userId={userId}
+          />
+        );
+      },
+      [userId]
+    );
```
Comment (RubenMCCarreira) [inline frontend/components/CardBoard/CardBody/CardBody.tsx:89]: I'm not sure if this is the correct way to render a JXS item. I think you should replace this for a new component

Comment (RubenMCCarreira) [reply frontend/components/CardBoard/CardBody/CardBody.tsx:89]: or try to join this some more logic in the existent component

**Pull request merged**